release/nightly_tests/dataset/inference.py

```diff
@@ -0,0 +1,143 @@
+# TODO (Alex): Split these into separate files once runtime envs work. The
+# releaser doesn't push files to worker nodes.
+################ model.py ###########################3
+from os import name
+import ray
+from ray import serve
+
+from io import BytesIO
+from PIL import Image
+import requests
+
+import torch
+from torchvision import transforms
+from torchvision.models import resnet50
+
+
+class Preprocessor:
+    def __init__(self):
+        self.torch_transform = transforms.Compose([
+            transforms.Resize(224),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Lambda(lambda t: t[:3, ...]),  # remove alpha channel
+            transforms.Normalize(
+                mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+        ])
+
+    def __call__(self, img_bytes):
+        try:
+            img = Image.open(BytesIO(img_bytes)).convert("RGB")
+            tensor = self.torch_transform(img)
+            return tensor
+        except Exception as e:
+            raise e
+
+
+class ImageModel:
+    def __init__(self):
+        self.model = resnet50(pretrained=True).eval().half().cuda()
+
+    def __call__(self, input_tensor_np):
+        input_tensor = torch.from_numpy(input_tensor_np).half().cuda()
+        with torch.no_grad():
+            output_tensor = self.model(input_tensor)
+            result = torch.argmax(output_tensor, dim=1).cpu()
+        return result.numpy()
+
+
+################# inference.py #######################
+
+import ray
+import boto3
+import json
+import requests
+import time
+import os
+from tqdm import tqdm
+import numpy as np
+
+
+def get_paths(bucket, path, max_files=100 * 1000):
```

Comment:
Can you move this to read binary files too.

Reply:
I don't think we want to.

1. It isn't not very general purpose (relies on a boto3 client that would typically have to be configured, does a weird cache, etc)
2. It's extremely slow

Reply:
Is it faster than the current path listing code? If so, move it there. Else, remove it and use the existing lister.

Reply:
what's the current code?

Reply:
I see, it seems read_binary_files() has no currently directory listing code, which is not satisfying the doc spec. Please add this so we can support passing directories to read_binary_files().